velox/expression/CastExpr-inl.h

```diff
@@ -296,6 +296,30 @@ VectorPtr CastExpr::applyDecimalToBooleanCast(
   return result;
 }
 
+template <typename FromNativeType>
+VectorPtr CastExpr::applyDecimalToVarcharCast(
+    const SelectivityVector& rows,
+    const BaseVector& input,
+    exec::EvalCtx& context,
+    const TypePtr& fromType) {
+  VectorPtr result;
+  context.ensureWritable(rows, VARCHAR(), result);
+  (*result).clearNulls(rows);
+  const auto simpleInput = input.as<SimpleVector<FromNativeType>>();
+  auto flatResult = result->asFlatVector<StringView>();
+
+  applyToSelectedNoThrowLocal(context, rows, result, [&](int row) {
+    if (simpleInput->isNullAt(row)) {
+      result->setNull(row, true);
+    } else {
+      std::string output =
+          DecimalUtil::toString((int128_t)simpleInput->valueAt(row), fromType);
+      flatResult->set(row, StringView(output));
```

Comment:
This may allocate buffers many times, I would suggest to compute the not inline string length and let `DecimalUtil::toString` can accept pre-allocated chat* out.
And it will introduce extra copy for not inlined StringView

Reply:
But I'm not sure the allocate times and memory copy cost compared to get decimal length cost by this strategy https://github.com/facebookincubator/velox/pull/6210#discussion_r1303687705, maybe we need a benchmark to check. I suppose the latter will be more effective for a vector with many rows.

Reply:
@jinchengchenghh The strategy to get length sounds good but may be not proper here. Because it requires `DecimalUtil::toString` always return decimal-based strings and we cannot guarantee that in code level.

Reply:
> This may allocate buffers many times, I would suggest to compute the not inline string length and let DecimalUtil::toString can accept pre-allocated chat* out.

Changed back to pre-allocate the buffer once.

Reply:
Please look into this discussion first, thanks! @mbasmanova